docs/ops/detection/Proposal_4.md

```diff
@@ -0,0 +1,194 @@
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: True
+  * **Required**: *no*
+
+* *clip_after_nms*
+
+  * **Description**: *clip_after_nms* is a flag that specifies whether to perform clip bounding boxes after non-maximum suppression or not.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+* *normalize*
+
+  * **Description**: *normalize* is a flag that specifies whether to perform normalization of output boxes to *[0,1]* interval or not.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+* *box_size_scale*
+
+  * **Description**: *box_size_scale* specifies the scale factor applied to box sizes before decoding.
+  * **Range of values**: a positive floating-point number
+  * **Type**: `float`
+  * **Default value**: 1.0
+  * **Required**: *no*
+
+* *box_coordinate_scale*
+
+  * **Description**: *box_coordinate_scale* specifies the scale factor applied to box coordinates before decoding.
+  * **Range of values**: a positive floating-point number
+  * **Type**: `float`
+  * **Default value**: 1.0
+  * **Required**: *no*
+
+* *framework*
+
+  * **Description**: *framework* specifies how the box coordinates are calculated.
+  * **Range of values**:
+    * "" (empty string) - calculate box coordinates like in Caffe*
+    * *tensorflow* - calculate box coordinates like in the TensorFlow* Object Detection API models
+  * **Type**: string
+  * **Default value**: "" (empty string)
+  * **Required**: *no*
+
+**Inputs**:
+
+*   **1**: 4D tensor of type *T* and shape `[batch_size, 2*K, H, W]` with class prediction scores. Required.
+
+*   **2**: 4D tensor of type *T* and shape `[batch_size, 4*K, H, W]` with deltas for each bounding box. Required.
+
+*   **3**: 1D tensor of type *T* with 3 or 4 elements:  `[image_height, image_width, scale_height_and_width]` or `[image_height, image_width, scale_height, scale_width]`. Required.
+
+**Outputs**4
+
+*   **1**: tensor of type *T* and shape `[batch_size * post_nms_topn, 5]`.
+
+*   **2**: tensor of type *T* and shape `[batch_size * post_nms_topn]` with probabilities. *Optional*.
+
+**Types**
+
+* *T*: floating point type.
+
+**Example**
+
+```xml
+<layer ... type="Proposal" ... >
+    <data base_size="16" feat_stride="8" min_size="16" nms_thresh="1.0" normalize="0" post_nms_topn="1000" pre_nms_topn="1000" ratio="1" scale="1,2"/>
+    <input>
+        <port id="0">
+            <dim>3</dim>
+            <dim>4</dim>
+            <dim>28</dim>
+            <dim>28</dim>
+        </port>
+        <port id="1">
+            <dim>3</dim>
+            <dim>8</dim>
+            <dim>28</dim>
+            <dim>28</dim>
+        </port>
+        <port id="2">
+            <dim>3</dim>
+        </port>
+    </input>
+    <output>
+        <port id="3" precision="FP32">
+            <dim>1000</dim>
```

Comment:
Should be 3000. The same for line 190.

Reply:
Actually, let's set the batch size to be 7 to not confuse with the shape of third input.

Reply:
Changed `batch_size` to 7.
>Should be 3000. 
Why? I don't understand

Reply:
This snippet of IR is from SSH network and it's `post_nms_topn` is equal to 1000. I advocate to leave it as it is.

Reply:
According to the specification the output shape is `[batch_size * post_nms_topn, 5]`

Reply:
🤦sorry for that. Updated.